packages/protocol/socket.ts

```diff
@@ -224,6 +235,28 @@ export function addEventListener<M extends EventMethods>(
   event: M,
   handler: (params: EventParams<M>) => void
 ) {
+  if (event === "Analysis.analysisPoints") {
+    gEventListeners.set(event, ({ analysisId, points }: analysisPoints) => {
+      const callbacks = gAnalysisCallbacks.get(analysisId);
+      if (callbacks) {
+        callbacks.onEvent(points);
+      } else {
+        handler({ analysisId, points });
+      }
+    });
```

Comment:
Why is this an if/else? Shouldn't we always call the `handler` _in addition to_ any callback that was registered? This seems unexpected to me.